subsys/bluetooth/shell/bt.c

```diff
@@ -1231,6 +1218,19 @@ static ssize_t ad_init(struct bt_data *data_array, const size_t data_array_size,
 		ad_len++;
 	}
 
+	if (appearance) {
+		const uint16_t appearance = bt_get_appearance();
+		static uint8_t appearance_data[2];
+
+		__ASSERT(data_array_size > ad_len, "No space for appearance");
```

Comment:
This condition needs to be handled. We don't want to crash the shell because of user error.